d2graph/d2graph.go

```diff
@@ -968,6 +968,18 @@ func (obj *Object) GetDefaultSize(mtexts []*d2target.MText, ruler *textmeasure.R
 	return &dims, nil
 }
 
+func (obj *Object) OuterNearContainer() *Object {
+	temp := obj
+	for temp != nil {
+		// find outer near container that isn't itself
```

Comment:
why exclude itself? doesn't that mean this code would let you connect two near containers?
https://github.com/terrastruct/d2/pull/1071/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-82cd6b1f1fa955a8d32bdac010cfbd436bc544dd3b445faf012100e8eb3f7a0dR773

Reply:
i tested and it's prevented somewhere else, but i still think there's no need to exclude itself

Reply:
i recall i was trying to fix a ci issue, is i don't exclude itself, `TestCompile/near_bad_connected` this testcase will throw   more that one error, is that normal and acceptable?
<img width="1279" alt="image" src="https://user-images.githubusercontent.com/19243114/229819223-43723a80-1805-4034-a694-6ea2a67ea894.png">

Reply:
multiple errors is normal, but in this case, only one of those errors should show up. but that's too disconnected from this code for me to think the proper fix is here

Reply:
so what about don't exclude itself, and change the expErr to 
`d2/testdata/d2compiler/TestCompile/near_bad_connected.d2:3:12: constant near keys cannot be set on connected shapes
d2/testdata/d2compiler/TestCompile/near_bad_connected.d2:5:5: cannot connect objects from within a container, that has near constant set, to objects outside that container`

Reply:
well, indeed, the second error doesn't make sense for this case, what about add a extra judgement when validate edges?

<img width="1352" alt="image" src="https://user-images.githubusercontent.com/19243114/229837143-2c0bcf5d-4cd4-49b3-a240-065e6b16c6f0.png">

Reply:
i think you can just get rid of the first error handle, as it'll always be handled by the second case right?

https://github.com/donglixiaoche/d2/blob/7a395ecf7175d1cd386cfe824c2e1daca20f8820/d2compiler/compile.go#L745

and then remove the `&&` you just added.

Reply:
done